File: unsorted/js/21.merge-two-sorted-lists.js
/*
 * @lc app=leetcode id=21 lang=javascript
 *
 * [21] Merge Two Sorted Lists
 */
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} l1
 * @param {ListNode} l2
 * @return {ListNode}
 */

function ListNode(val) {
	this.val = val;
	this.next = null;
}

function arrayToList(array) {
	let dummy = new ListNode(0);
	let node = dummy;
	array.forEach((a) => {
		node.next = new ListNode(a);
		node = node.next;
	});
	return dummy.next;
}

var mergeTwoLists = function(l1, l2) {
	let dummy = new ListNode(0);
	let node = dummy;
	while (l1 && l2) {
		if (l1.val <= l2.val) {
			node.next = l1;
			l1 = l1.next;
		} else {
			node.next = l2;
			l2 = l2.next;
		}
		node = node.next;
	}
	if (l1) {
		node.next = l1;
	}
	if (l2) {
		node.next = l2;
	}

	return dummy.next;
};

File: unsorted/java/21.merge-two-sorted-lists.java
/*
 * @lc app=leetcode id=21 lang=java
 *
 * [21] Merge Two Sorted Lists
 */

// @lc code=start
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode mergeTwoLists(ListNode l1, ListNode l2) {
        ListNode dummy = new ListNode(0);
        ListNode p = dummy;
        while(l1 != null && l2 != null){

            if(l1.val < l2.val){
                p.next = l1;
                l1 = l1.next;
            }else{
                p.next = l2;
                l2 = l2.next;
            }
            p = p.next;
        }
        while(l1 != null){
            p.next = l1;
            l1 = l1.next;
            p = p.next;
        }
        while(l2 != null){
            p.next = l2;
            l2= l2.next;
            p = p.next;
        }
        return dummy.next;
        
    }
}
// @lc code=end

